cartography/intel/semgrep/findings.py

```diff
@@ -0,0 +1,217 @@
+        sca_vuln["title"] = vuln["advisory"]["title"]
+        sca_vuln["description"] = vuln["advisory"]["description"]
+        sca_vuln["ecosystem"] = vuln["advisory"]["ecosystem"]
+        sca_vuln["severity"] = vuln["advisory"]["severity"]
+        sca_vuln["reachability"] = vuln["advisory"]["reachability"]
+        sca_vuln["reachableIf"] = vuln["advisory"]["reachableIf"]
+        sca_vuln["exposureType"] = vuln["exposureType"]
+        dependency = f"{vuln['matchedDependency']['name']}|{vuln['matchedDependency']['versionSpecifier']}"
+        sca_vuln["matchedDependency"] = dependency
+        sca_vuln["dependencyFileLocation_path"] = vuln["dependencyFileLocation"]["path"]
+        sca_vuln["dependencyFileLocation_url"] = vuln["dependencyFileLocation"]["url"]
+        # Optional fields
+        cves = vuln.get("advisory", {}).get("references", {}).get("cveIds")
+        if len(cves) > 0:
+            # Take the first CVE
+            sca_vuln["cveId"] = vuln["advisory"]["references"]["cveIds"][0]
+        if vuln.get('closestSafeDependency'):
+            dep_fix = f"{vuln['closestSafeDependency']['name']}|{vuln['closestSafeDependency']['versionSpecifier']}"
+        else:
+            dep_fix = None
+        sca_vuln["closestSafeDependency"] = dep_fix
+        ref_urls = vuln["advisory"].get("references", {}).get("urls", [])
+        if ref_urls:
+            sca_vuln["ref_urls"] = ",".join(ref_urls)
+        sca_vuln["openedAt"] = vuln.get("openedAt", None)
+        usages_list = []
+        for usage in vuln.get("usages", []):
+            usage_dict = {}
+            usage_dict["findingId"] = usage["findingId"]
+            usage_dict["path"] = usage["location"]["path"]
+            usage_dict["startLine"] = usage["location"]["startLine"]
+            usage_dict["startCol"] = usage["location"]["startCol"]
+            usage_dict["endLine"] = usage["location"]["endLine"]
+            usage_dict["endCol"] = usage["location"]["endCol"]
+            usage_dict["url"] = usage["location"]["url"]
+            usages_list.append(usage_dict)
+        sca_vuln["usages"] = usages_list
+        vulns.append(sca_vuln)
+    return vulns
+
+
+@timeit
+def load_semgrep_deployment(
+    neo4j_session: neo4j.Session, deployment: List[Dict[str, Any]], update_tag: int,
+) -> None:
+    logger.info(f"Loading Semgrep deployment info {deployment} into the graph...")
+    load(
+        neo4j_session,
+        SemgrepDeploymentSchema(),
+        deployment,
+        lastupdated=update_tag,
+    )
+
+
+@timeit
+def load_semgrep_sca_vulns(
+    neo4j_session: neo4j.Session,
+    vulns: List[Dict[str, Any]],
+    deployment_id: str,
+    update_tag: int,
+) -> None:
+    logger.info(f"Loading {len(vulns)} Semgrep SCA vulns info into the graph.")
+    load(
+        neo4j_session,
+        SemgrepSCAFindingSchema(),
+        vulns,
+        lastupdated=update_tag,
+        DEPLOYMENT_ID=deployment_id,
+    )
+    for vuln in vulns:
```

Comment:
For better performance, all the usages should be extracted out first, so they can all be loaded once.

Reply:
Moved it to it's own load function.